gen/src/generate/idna/shared/version.rs

```diff
@@ -0,0 +1,55 @@
+// TODO: Unify this with generate::ucd::shared::version
```

Comment:
Here I meant to refactor specifically `unic-gen::generate::idna::shared::version` and `unic-gen::generate::ucd::shared::version` to use the same code. They're basically the same module, just with a single string difference of which data file they're looking in. I could and probably should make them one shared file which I give the file handle or path to, rather than duplicating the file.

Of course, having the ReadMe file in the data be for the resource its downloaded and the version included in the components tables the version that its tables were generated from is what should be kept.